src/game/client/components/maplayers.cpp

```diff
@@ -531,7 +531,7 @@ void CMapLayers::BackgroundMapUpdate()
 	{
 		// unload map
 		m_pMenuMap->Unload();
-
-		LoadBackgroundMap();
+		if(Config()->m_ClShowMenuMap)
+			LoadBackgroundMap();
```

Comment:
This `if` should not be useful, as `LoadBackgroundMap` starts with

```cpp
	if(!Config()->m_ClShowMenuMap)
		return;
```

Reply:
I also removed the `if` from the `LoadBackgroundMap` method itself, because I'd have to check it outside the method anyway in the case below.